### PR TITLE
News Posts In its own Directory

### PR DESCRIPTION
**Pull request by Heinrich-XIAO**

### CODE REVIEW

**Comment by Heinrich-XIAO** [comment]
#250

**Comment by Naviary3** [comment]
Great start! Here's how I'm imagining this would work:

We have a news post folder, followed by directories of each language:
news/
  en-US/
    2024-8-1.txt
  fr-FR/
    2024-8-1.txt

The news posts would be written in markdown language. This allows us to format text using plain text. For example, `2024-8-1.txt` would look like:

```
Update 1.4 is released! There have been many collaborative features added since we open sourced!

* Knightriders have been added, which hop infinitely like a knight until they're obstructed! The 'Knighted Chess' variant has been upgraded to replace the knights with knightriders!
* Click your or your opponent's pieces at any time to view their possible moves!
* Right-click at any time to deselect the currently selected piece.
* Hovering over arrow indicators on the edge of the screen now renders the legal moves of the piece they are pointing to!
* The game now automatically declares a draw if there's insufficient material on the board to force checkmate.
* Translated the website into French! You can change the language by visiting the footer on any page.
* Improved the loading time of the website
* New website icon, Ω! This automatically matches your preferred light or dark device theme.
* The game code's, or the ICN's, metadata has been reformatted to more closely match PGN norms.
* Users can now delete their account on their profile page, if they so choose, without having to email us.
```

In markdown, the `*` is auto-replaced with bullet points. So no coding or html is required when we write or translate the news posts! We also have the option of using `#` to signify headings. There's a lot of formatting options at our disposal.

Then, I imagine that on server startup, when it prepares the html files for serving, it would read through each news post in the **english** news directory, check if the language-specific translation is present for it, if so it adds it to the html, otherwise it adds the english news post to the html.

We can use an npm package like [marked](https://www.npmjs.com/package/marked) which will convert the text files written in markdown into html with the appropriate tags obtain the specified formatting.

**Comment by Naviary3** [comment]
This will also shorten all the TOML files by 85 lines!

**Comment by Heinrich-XIAO** [comment]
but things like the iframe wont work

**Comment by Heinrich-XIAO** [comment]
so, should i do html  files? also, having dates for the file names is not very easy to order/parse. so, should the file be marked in the metadata with a specific order?

**Comment by Heinrich-XIAO** [comment]
.md or .txt?

**Comment by Naviary3** [comment]
As for no iframe support, we could use a workaround for the `marked` package, that will detect youtube links and insert an iframe:
https://chatgpt.com/share/66f33fed-d1e8-8011-9355-99bc0f6cfdca

**Comment by Naviary3** [comment]
> .md or .txt?

.md is actually probably better, since it stands for markdown

**Comment by Naviary3** [comment]
> having dates for the file names is not very easy to order/parse. so, should the file be marked in the metadata with a specific order?

I can't imagine it that hard. We store the filename as the date an time in ISO 8601 Format: `YYYY-MM-DDTHH-MM-SS`, we can get a list of all files present in a directory, and I bet ChatGPT could easily write a method for iterating through the files in order according to the format.

**Comment by Heinrich-XIAO** [comment]
I have made spaghetti. 
line 219
```js					
news: newsFiles.map(filePath => marked.parse(fs.existsSync(path.join(folder, 'news', languageCode, filePath)) ? fs.readFileSync(path.join(folder, 'news', languageCode, filePath)).toString() : fs.readFileSync(path.join(folder, 'news', getDefaultLanguage(), filePath)).toString())).join('\n<br>\n') // I will refactor this later
```

**Comment by Heinrich-XIAO** [comment]
should there be a br between the time and the content?

Not from my pr.
image

**Comment by Naviary2** [comment]
> should there be a br between the time and the content?

I would probably prefer a bar between news posts themselves. I may end up tweaking the whole styling, like having the date be auto-placed at the top of each news post, in small, grey text, instead of manually putting it in the post. I don't expect you to do that before this is merge-ready though.

**Comment by Heinrich-XIAO** [comment]
how would the date be auto placed for translations

**Comment by Heinrich-XIAO** [comment]
> > should there be a br between the time and the content?
> 
> I would like a visible line between news posts themselves. I may end up tweaking the whole styling, like having the date be auto-placed at the top of each news post, in small, grey text, instead of manually putting it in the post. I don't expect you to do that before this is merge-ready though.

So, an hr between the date and the content or as a seperator?

**Comment by Naviary2** [comment]
> So, an hr between the date and the content or as a seperator?

An <hr> is what I had in mind for separating each post (and nothing between the date and text), but it will probably only look good if the date of each post is small and grey

**Comment by Naviary2** [comment]
> how would the date be auto placed for translations

Take the name of the news post file, which should be the date in ISO 8601 format, and use the already-acquired date-fns package to convert the date into english language: https://chatgpt.com/share/66f70a4e-2530-8004-a43c-27450c9a5c4f

Then add that to the top of the news post, with its own unique styling. I'd probably make it a paragraph with a "date" class, which in the css will make it small and grey.

**Comment by Heinrich-XIAO** [comment]
Chad Jippity's code didn't work at least with est. now i have to fix it.

**Comment by Heinrich-XIAO** [comment]
Ready to review.

File: translation/en-US.toml [hunk 1]
```diff
@@ -483,91 +483,6 @@ login_button = "Log In"
 409_message = ["There may have been a clashing username or email. Please ", "reload", ", the page."]
 500_message = "This isn't supposed to happen. There is some debugging to be done!"
 
-########### NEWS ###########
-
-[news]
-title = "News" # The tab name
-more_dev_logs = ["More dev logs are posted on the ", "official discord", ", and on the ", "chess.com forums!"]
```
Comment (Naviary3) [inline translation/en-US.toml:490]: Oh actually we should keep `title` and `more_dev_logs`.

Comment (Heinrich-XIAO) [reply translation/en-US.toml:490]: done

**Comment by Naviary2** [comment]
I didn't get to reviewing this today. I'll plan to tomorrow!

**Comment by Heinrich-XIAO** [comment]
just fixed all the other indentation issues.

**Pull request merged**